datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -72,6 +74,15 @@ pub enum AggregateMode {
     Single,
 }
 
+/// Group By expression modes
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum GroupByOrderMode {
+    /// Some of the expressions in the GROUP BY clause have an ordering.
+    PartiallyOrdered,
```

Comment:
`PartiallyOrdered` is left over for follow-up PRs, right?

Reply:
Algorithm for finding different group boundaries are different for the `Ordered` and `PartiallyOrdered` cases. We could have used a flag instead of this struct. However, I think this is more explicit.